clone/internal/download/http.go

```diff
@@ -0,0 +1,53 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package download
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+)
+
+// NewHTTPFetcher returns an HTTPFetcher that gets paths appended to the given prefix.
+func NewHTTPFetcher(prefix string) HTTPFetcher {
+	return HTTPFetcher{
+		baseURL: prefix,
+	}
+}
+
+// HTTPFetcher gets the data over HTTP(S).
+// This is thread safe.
+type HTTPFetcher struct {
+	baseURL string
+}
+
+// GetData gets the data at the given path.
+func (f HTTPFetcher) GetData(path string) ([]byte, error) {
+	target := f.baseURL + path
```

Comment:
Worth using `url.URL` to avoid brittle bits with paths/slashes etc.?
(separately might want to enforce that the user provided flag has a trailing slash?)